File: backend/src/main/kotlin/mymoney/domain/repository/GoalRepository.kt
package mymoney.domain.repository

import kotlinx.datetime.Instant
import mymoney.domain.model.Goal
import java.util.UUID

interface GoalRepository {
    suspend fun create(goal: Goal): Goal

    suspend fun findById(id: UUID): Goal?

    suspend fun list(familyId: UUID): List<Goal>

    suspend fun update(goal: Goal): Goal

    suspend fun softDelete(id: UUID, now: Instant)
}

File: mobile/lib/domain/repository/goal_repository.dart
import '../model/goal.dart';

abstract class GoalRepository {
  Future<void> create(Goal goal);
  Future<Goal?> findById(String id);
  Future<List<Goal>> listByFamily(String familyId);
  Future<void> update(Goal goal);
  Future<void> softDelete(String id);
  Stream<List<Goal>> watchByFamily(String familyId);
}
